### PR TITLE
DOC: Solve Dependencies Conflicts and pyproject build

### PR DESCRIPTION
**Pull request by phmbressan**
## Pull request type

- [X] Code changes (bugfix, features)
- [X] ReadMe, Docs and GitHub updates

## Checklist

- [X] Docs have been reviewed and added / updated
- [X] Lint (`black rocketpy/ tests/`) has passed locally 
- [ ] All tests (`pytest tests -m slow --runslow`) have passed locally
- [x] `CHANGELOG.md` has been updated (if relevant)

Note: `--runslow` is currently failing.

## Current behavior

An error occurs while trying building docs due to many reasons, including:

- New release of `docutils` not supported by `m2r2`;
- `pyproject.toml` requires `conf.py` changes;
- `lxml` separated its `html_clean` to another repo for security.

All those library changes happened around late april and beginning of may.

## New behavior

The following changes were made:

- Limit `docutils` version to `0.20.1`;
- Add to `docs/requirements` the new `lxml_html_clean`;
- Add `PYTHONPATH` to `conf.py`.

Following [sphinx docs recommendation](https://docs.readthedocs.io/en/stable/guides/reproducible-builds.html#pin-your-transitive-dependencies), in order to avoid breaking docs due to dependency updates `pip-tools` was used to pin down all docs dependencies. 

The following procedure should be followed to add/update a docs dependency:

1. Add the new top-level dependency to `docs/requirements.in`;
2. Run `pip-compile docs/requirements.in`.

<details>
  <summary>Docs are building now:</summary>
  image
</details>

## Breaking change

- [ ] Yes
- [X] No

## Additional information

The PR #584 removes `README.md` from the docs, allowing the removal of `m2r2` dependency and the limitation of `docutils` version.

### CODE REVIEW

File: .readthedocs.yaml [hunk 1]
```diff
@@ -7,9 +7,9 @@ version: 2
 
 # Set the version of Python and other tools
 build:
-  os: ubuntu-20.04
+  os: ubuntu-22.04
```
Comment (Gui-FernandesBR) [inline .readthedocs.yaml:10]: Did you consider using `ubuntu-latest` instead? 

I'm afraid that we will need to update this in a few years again, no?

All in all, this doesn't prevent us from merging this PR.

File: docs/requirements.in
Comment (Gui-FernandesBR) [inline docs/requirements.in:1]: Interesting to mention that you have updated most of the requirements for the docs.
This is something good.

**Review by Gui-FernandesBR — APPROVED** [review]
Really nice PR, thanks for the solution.

Would you mind adding a comment to ~ somewhere I don't know ~ saying that you need to run `pip-compile ... ` to generate the requirements.txt file now?

Maybe at the top of the `requirements.in` ?

All tests are passing in the CI. I'm approving the PR, can be merged already I think.

**Comment by phmbressan** [comment]
There are some comments added automatically at the top of `docs/requirements.txt` which specify what you mentioned.

I don't know if you have seen those, but if you think that it is not enough, I can add a line on top of `requirements.in` with the same.

**Comment by Gui-FernandesBR** [comment]
> There are some comments added automatically at the top of `docs/requirements.txt` which specify what you mentioned.
> 
> I don't know if you have seen those, but if you think that it is not enough, I can add a line on top of `requirements.in` with the same.

OK, Good, that's enough, I haven't seen those before.
All good.

File: .readthedocs.yaml (hunk 1 above)
Comment (phmbressan) [reply .readthedocs.yaml:10]: I did't consider it, but I prefer keeping the way it is. I my experience, when Ubuntu updates its version, there might be some instabilities with packages that are migrating or made breaking changes.

Updating the os is important, but it happens so infrequently that I believe doing it manually makes more sense for us.

**Pull request merged**